### PR TITLE
`azurerm_data_factory_integration_runtime_azure_ssis` - support Elastic Pool

### LINKED ISSUE #18079: azurerm_data_factory_integration_runtime_azure_ssis catalog database service tier is not accepting elastic pool
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.7

### AzureRM Provider Version

3.1.0

### Affected Resource(s)/Data Source(s)

azurerm_data_factory_integration_runtime_azure_ssis

### Terraform Configuration Files

```hcl
resource "azurerm_data_factory_integration_runtime_azure_ssis" "multiple" {
  count                            = length(var.integration_runtimes_ssis)
  name                             = var.integration_runtimes_ssis[count.index].name
  data_factory_id                  = azurerm_data_factory.this.id
  location                         = var.location
  node_size                        = var.integration_runtimes_ssis[count.index].node_size
  number_of_nodes                  = var.integration_runtimes_ssis[count.index].number_of_nodes
  max_parallel_executions_per_node = var.integration_runtimes_ssis[count.index].max_parallel_executions_per_node
  edition                          = var.integration_runtimes_ssis[count.index].edition

  catalog_info {
    server_endpoint        = "${var.integration_runtimes_ssis[count.index].catalog.sql_server_name}.database.windows.net"
    administrator_login    = var.integration_runtimes_ssis[count.index].catalog.sql_server_admin_username
    administrator_password = var.integration_runtimes_ssis[count.index].catalog.sql_server_admin_password
    pricing_tier           = var.integration_runtimes_ssis[count.index].catalog.pricing_tier
  }
}
```

### Debug Output/Panic Output

```shell
Error: expected catalog_info.0.pricing_tier to be one of [Basic S0 S1 S2 S3 S4 S6 S7 S9 S12 P1 P2 P4 P6 P11 P15 GP_S_Gen5_1 GP_S_Gen5_2 GP_S_Gen5_4 GP_S_Gen5_6 GP_S_Gen5_8 GP_S_Gen5_10 GP_S_Gen5_12 GP_S_Gen5_14 GP_S_Gen5_16 GP_S_Gen5_18 GP_S_Gen5_20 GP_S_Gen5_24 GP_S_Gen5_32 GP_S_Gen5_40 GP_Gen5_2 GP_Gen5_4 GP_Gen5_6 GP_Gen5_8 GP_Gen5_10 GP_Gen5_12 GP_Gen5_14 GP_Gen5_16 GP_Gen5_18 GP_Gen5_20 GP_Gen5_24 GP_Gen5_32 GP_Gen5_40 GP_Gen5_80 BC_Gen5_2 BC_Gen5_4 BC_Gen5_6 BC_Gen5_8 BC_Gen5_10 BC_Gen5_12 BC_Gen5_14 BC_Gen5_16 BC_Gen5_18 BC_Gen5_20 BC_Gen5_24 BC_Gen5_32 BC_Gen5_40 BC_Gen5_80 HS_Gen5_2 HS_Gen5_4 HS_Gen5_6 HS_Gen5_8 HS_Gen5_10 HS_Gen5_12 HS_Gen5_14 HS_Gen5_16 HS_Gen5_18 HS_Gen5_20 HS_Gen5_24 HS_Gen5_32 HS_Gen5_40 HS_Gen5_80], got ElasticPool
```

### Expected Behaviour

As supported in the portal, it would be great to have ElasticPool supported in TF. 

### Actual Behaviour

We cannot use ElasticPool for creating a Catalog database for SSIS runtime.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by ziyeqf**
fix #18079

test
---
```
TF_ACC=1 go test -v ./internal/services/datafactory -run=TestAccDataFactoryIntegrationRuntimeManagedSsis_basic 
=== RUN   TestAccDataFactoryIntegrationRuntimeManagedSsis_basic
=== PAUSE TestAccDataFactoryIntegrationRuntimeManagedSsis_basic
=== CONT  TestAccDataFactoryIntegrationRuntimeManagedSsis_basic
--- PASS: TestAccDataFactoryIntegrationRuntimeManagedSsis_basic (255.56s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory   256.664s

TF_ACC=1 go test -v ./internal/services/datafactory -run=TestAccDataFactoryIntegrationRuntimeManagedSsis_withElasticPool    
=== RUN   TestAccDataFactoryIntegrationRuntimeManagedSsis_withElasticPool
=== PAUSE TestAccDataFactoryIntegrationRuntimeManagedSsis_withElasticPool
=== CONT  TestAccDataFactoryIntegrationRuntimeManagedSsis_withElasticPool
--- PASS: TestAccDataFactoryIntegrationRuntimeManagedSsis_withElasticPool (423.17s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/datafactory   424.286s

```

### CODE REVIEW

File: internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource.go [hunk 1]
```diff
@@ -230,6 +232,11 @@ func resourceDataFactoryIntegrationRuntimeAzureSsis() *pluginsdk.Resource {
 								"HS_Gen5_2", "HS_Gen5_4", "HS_Gen5_6", "HS_Gen5_8", "HS_Gen5_10", "HS_Gen5_12", "HS_Gen5_14", "HS_Gen5_16", "HS_Gen5_18", "HS_Gen5_20", "HS_Gen5_24", "HS_Gen5_32", "HS_Gen5_40", "HS_Gen5_80",
 							}, false),
 						},
+						"elastic_pool_name": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: sqlValidate.ValidateMsSqlElasticPoolName,
+						},
```
Comment (magodo) [inline internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource.go:239]: Can we add an conflict with for the `elastic_pool_name` and `pricing_tier` as they both correspond to the same API property?

File: internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource.go [hunk 2]
```diff
@@ -638,9 +645,16 @@ func expandDataFactoryIntegrationRuntimeAzureSsisProperties(d *pluginsdk.Resourc
 	if catalogInfos, ok := d.GetOk("catalog_info"); ok && len(catalogInfos.([]interface{})) > 0 {
 		catalogInfo := catalogInfos.([]interface{})[0].(map[string]interface{})
 
+		var pricingTier datafactory.IntegrationRuntimeSsisCatalogPricingTier
```
Comment (magodo) [inline internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource.go:648]: Can we add some comments about the fact that both `pricing_tier` and `elastic_pool_name` ends up to the same API prop: `CatalogPricingTier`

File: internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource.go [hunk 3]
```diff
@@ -869,6 +883,15 @@ func flattenDataFactoryIntegrationRuntimeAzureSsisCatalogInfo(ssisProperties *da
 		dualStandbyPairName = *ssisProperties.DualStandbyPairName
 	}
 
+	var pricingTier, elasticPoolName string
+	if strings.HasPrefix(string(ssisProperties.CatalogPricingTier), "ELASTIC_POOL") {
+		if result := regexp.MustCompile(`(?:name=\")([^&%\\\/?]{0,127}[^\s.&%\\\/?])(?:\")`).FindStringSubmatch(string(ssisProperties.CatalogPricingTier)); len(result) > 1 {
+			elasticPoolName = result[1]
+		}
```
Comment (magodo) [inline internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource.go:890]: Can we put this piece of code to a separate function (like `parseElasticPool()`)? Also we shall simplify it to something like below:

```
{
    matches := regexp.MustCompile(`^ELASTIC_POOL\(name="(.+)"\)$`).FindStringSubmatch(s)
    if len(matches) != 2 {
      return "", false
   }
   return matches[1], true
}
```

File: internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource.go [hunk 4]
```diff
@@ -1111,3 +1135,7 @@ func readBackSensitiveValue(input []interface{}, propertyName string, filters ma
 	}
 	return ""
 }
+
+func expandDataFactoryIntegrationRuntimeElasticPool(input string) string {
```
Comment (magodo) [inline internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource.go:1139]: Can we rename this function to `formatDataFactoryIntegrationRuntimeElasticPool`?

File: website/docs/r/data_factory_integration_runtime_azure_ssis.html.markdown [hunk 5]
```diff
@@ -79,7 +79,9 @@ A `catalog_info` block supports the following:
...
+
+* `elastic_pool_name` - (Optional) the name of SQL elastic pool where the database will be created for the SSIS catalog. (Mutually exclusive with `pricing_tier`.)
```
Comment (magodo) [inline website/docs/r/data_factory_integration_runtime_azure_ssis.html.markdown:84]:
```suggestion
* `elastic_pool_name` - (Optional) the name of SQL elastic pool where the database will be created for the SSIS catalog. Mutually exclusive with `pricing_tier`.
```

File: website/docs/r/data_factory_integration_runtime_azure_ssis.html.markdown [hunk 6]
```diff
@@ -79,7 +79,9 @@ A `catalog_info` block supports the following:
...
+* `pricing_tier` - (Optional) Pricing tier for the database that will be created for the SSIS catalog. Valid values are: `Basic`, `S0`, `S1`, `S2`, `S3`, `S4`, `S6`, `S7`, `S9`, `S12`, `P1`, `P2`, `P4`, `P6`, `P11`, `P15`, `GP_S_Gen5_1`, `GP_S_Gen5_2`, `GP_S_Gen5_4`, `GP_S_Gen5_6`, `GP_S_Gen5_8`, `GP_S_Gen5_10`, `GP_S_Gen5_12`, `GP_S_Gen5_14`, `GP_S_Gen5_16`, `GP_S_Gen5_18`, `GP_S_Gen5_20`, `GP_S_Gen5_24`, `GP_S_Gen5_32`, `GP_S_Gen5_40`, `GP_Gen5_2`, `GP_Gen5_4`, `GP_Gen5_6`, `GP_Gen5_8`, `GP_Gen5_10`, `GP_Gen5_12`, `GP_Gen5_14`, `GP_Gen5_16`, `GP_Gen5_18`, `GP_Gen5_20`, `GP_Gen5_24`, `GP_Gen5_32`, `GP_Gen5_40`, `GP_Gen5_80`, `BC_Gen5_2`, `BC_Gen5_4`, `BC_Gen5_6`, `BC_Gen5_8`, `BC_Gen5_10`, `BC_Gen5_12`, `BC_Gen5_14`, `BC_Gen5_16`, `BC_Gen5_18`, `BC_Gen5_20`, `BC_Gen5_24`, `BC_Gen5_32`, `BC_Gen5_40`, `BC_Gen5_80`, `HS_Gen5_2`, `HS_Gen5_4`, `HS_Gen5_6`, `HS_Gen5_8`, `HS_Gen5_10`, `HS_Gen5_12`, `HS_Gen5_14`, `HS_Gen5_16`, `HS_Gen5_18`, `HS_Gen5_20`, `HS_Gen5_24`, `HS_Gen5_32`, `HS_Gen5_40` and `HS_Gen5_80`. (Mutually exclusive with `elastic_pool_name`.)
```
Comment (magodo) [inline website/docs/r/data_factory_integration_runtime_azure_ssis.html.markdown:82]:
```suggestion
* `pricing_tier` - (Optional) Pricing tier for the database that will be created for the SSIS catalog. Valid values are: `Basic`, `S0`, `S1`, `S2`, `S3`, `S4`, `S6`, `S7`, `S9`, `S12`, `P1`, `P2`, `P4`, `P6`, `P11`, `P15`, `GP_S_Gen5_1`, `GP_S_Gen5_2`, `GP_S_Gen5_4`, `GP_S_Gen5_6`, `GP_S_Gen5_8`, `GP_S_Gen5_10`, `GP_S_Gen5_12`, `GP_S_Gen5_14`, `GP_S_Gen5_16`, `GP_S_Gen5_18`, `GP_S_Gen5_20`, `GP_S_Gen5_24`, `GP_S_Gen5_32`, `GP_S_Gen5_40`, `GP_Gen5_2`, `GP_Gen5_4`, `GP_Gen5_6`, `GP_Gen5_8`, `GP_Gen5_10`, `GP_Gen5_12`, `GP_Gen5_14`, `GP_Gen5_16`, `GP_Gen5_18`, `GP_Gen5_20`, `GP_Gen5_24`, `GP_Gen5_32`, `GP_Gen5_40`, `GP_Gen5_80`, `BC_Gen5_2`, `BC_Gen5_4`, `BC_Gen5_6`, `BC_Gen5_8`, `BC_Gen5_10`, `BC_Gen5_12`, `BC_Gen5_14`, `BC_Gen5_16`, `BC_Gen5_18`, `BC_Gen5_20`, `BC_Gen5_24`, `BC_Gen5_32`, `BC_Gen5_40`, `BC_Gen5_80`, `HS_Gen5_2`, `HS_Gen5_4`, `HS_Gen5_6`, `HS_Gen5_8`, `HS_Gen5_10`, `HS_Gen5_12`, `HS_Gen5_14`, `HS_Gen5_16`, `HS_Gen5_18`, `HS_Gen5_20`, `HS_Gen5_24`, `HS_Gen5_32`, `HS_Gen5_40` and `HS_Gen5_80`. Mutually exclusive with `elastic_pool_name`.
```

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I've taken a look through and left some comments inline, but this is mostly looking good to me 👍

**Review by katbyte — APPROVED** [review]
LGTM 🏷️

**Pull request merged**